### PR TITLE
-Added Scene service provider for Editor plugins

### PR DESCRIPTION
**Pull request by JohnnyWahib**
-Added UI service functions

### CODE REVIEW

**Comment by JoshEngebretson** [comment]
Thanks @JohnnyWahib, looks solid!  Hey @shaddockh would you mind having a once over on this PR with any TS/Plugin thoughts?

**Comment by shaddockh** [comment]
Sure thing, I'll look it over.

File: Script/AtomicEditor/hostExtensions/HostExtensionServices.ts [hunk 1]
```diff
@@ -206,21 +270,13 @@ export class UIServicesProvider extends ServicesProvider<Editor.HostExtensions.U
...
     }
 
-    private mainFrameMenu: MainFramMenu = null;
-    private hierarchyFrameMenu: HierarchyFrameMenu = null;
-    private projectFrameMenu: ProjectFrameMenu = null;
+    private mainFrame: MainFrame = null;
     private modalOps: ModalOps;
 
-    init(mainFrameMenu: MainFramMenu, hierarchyFrameMenu: HierarchyFrameMenu, projectFrameMenu: ProjectFrameMenu, modalOps: ModalOps) {
+    init(mainFrame: MainFrame, modalOps: ModalOps) {
```
Comment (shaddockh) [inline Script/AtomicEditor/hostExtensions/HostExtensionServices.ts]: nice cleanup of the signature.  No need to pass all those elements in if they are all available on the main frame.

File: Script/TypeScript/EditorWork.d.ts [hunk 2]
```diff
@@ -266,7 +276,11 @@ declare module Editor.HostExtensions {
         removeHierarchyContextMenuItemSource(id: string);
         createProjectContextMenuItemSource(id: string, items: any): Atomic.UIMenuItemSource;
         removeProjectContextMenuItemSource(id: string);
+        refreshHierarchyFrame();
         showModalWindow(windowText: string, uifilename: string, handleWidgetEventCB: (ev: Atomic.UIWidgetEvent) => void): Editor.Modal.ExtensionWindow;
+        showModalError(windowText: string, message: string);
+        showResourceSelection(windowText: string, importerType: string, resourceType: string, callback: (retObject: any, args: any) => void, args?: any);
```
Comment (shaddockh) [inline Script/TypeScript/EditorWork.d.ts]: This is a nice addition!

**Comment by shaddockh** [comment]
This all looks pretty good.  It does open up a thought I was just having last night about the need to expose standard editor dialogs to the extensions.  In addition to the error dialog, I could also see having access to an information dialog as well as some kind of progress bar.  Not for this PR, but something to think about.

Good job!

**Comment by JoshEngebretson** [comment]
Excellent, thanks guys!  LANDING!

**Pull request merged**